### PR TITLE
Olympia: Improved handling of proposal params through json config

### PR DESCRIPTION
**Pull request by mnaamani**
Fixes https://github.com/Joystream/joystream/issues/3273

1 - Catch invalid JSON when running tests.
2 - ~Fallback to default values instead of `panic!` in runtime. (panic only in unit tests)~
3 - Empty string and unset variable treated as no explicit JSON config is being passed, and defaults are used.
4 - Pass ALL_PROPOSALS_PARAMETERS_JSON as build arg to dockerfile

### CODE REVIEW

File: runtime/src/proposals_configuration/mod.rs [hunk 1]
```diff
@@ -109,12 +109,24 @@ lazy_static! {
 fn get_all_proposals_parameters_objects() -> AllProposalsParameters {
     let json_str: Option<&'static str> = option_env!("ALL_PROPOSALS_PARAMETERS_JSON");
 
+    // Handle undefined variable (null) and variable set to empty string the same
+    // to work cross platform.
+    if json_str.is_none() || json_str.unwrap().is_empty() {
```
Comment (shamil-gadelshin) [inline runtime/src/proposals_configuration/mod.rs]: The `unwap()` method should be avoided in the production code. The first part of the condition `.is_none()` could be modified and we would end up with panic. Using the `map_or` looks like a safer approach.

File: runtime/src/proposals_configuration/mod.rs [hunk 2]
```diff
@@ -109,12 +109,24 @@ lazy_static! {
...
+        return default_parameters();
+    }
+
     json_str
         .map(lite_json::parse_json)
         .map(|res| match res {
             Ok(json) => Some(json),
             Err(_) => {
-                panic!("Invalid JSON with proposals parameters provided.");
+                if cfg!(test) {
```
Comment (shamil-gadelshin) [inline runtime/src/proposals_configuration/mod.rs]: Why do we allow invalid JSON in non-test cases?

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]
Looks good. There are a couple of minor comments. It looks like there are clippy issues as well present in the code.

I corrupted the `./tests/integration-tests/proposal-parameters.json` file, ran the `./build-node-docker.sh`, and expected the build to fail but it was OK.

File: runtime/src/proposals_configuration/mod.rs (hunk 2 above)
Comment (mnaamani) [reply runtime/src/proposals_configuration/mod.rs]: I was trying to follow the description of the expected behavior in https://github.com/Joystream/joystream/blob/8bb0a84bc822eb817401e8ddd0f00e1613a40ccb/runtime/src/proposals_configuration/mod.rs#L6

Specifically "If it doesn't exists or contains invalid or empty JSON then the default parameters are returned."

**Comment by mnaamani** [comment]
> Looks good. There are a couple of minor comments. It looks like there are clippy issues as well present in the code.
> 

The clippy warning are because I commented out the proposals_integration module in `runtime/src/tests/mod.rs` but didn't comment out some common code which was being used from those tests.

> I corrupted the `./tests/integration-tests/proposal-parameters.json` file, ran the `./build-node-docker.sh`, and expected the build to fail but it was OK.

Did you run with `TEST_NODE=true ./build-node-docker.sh` ?

File: runtime/src/proposals_configuration/mod.rs (hunk 2 above)
Comment (mnaamani) [reply runtime/src/proposals_configuration/mod.rs]: But if we want compile time and runtime behavior to be the same, I can drop the condition `cfg!(test)` and always return `None` here.

File: runtime/src/proposals_configuration/mod.rs (hunk 1 above)
Comment (mnaamani) [reply runtime/src/proposals_configuration/mod.rs]: Fixed in 565dd4f536bb879de3dad829fa64656f7e98f7a3

File: runtime/src/proposals_configuration/mod.rs (hunk 2 above)
Comment (mnaamani) [reply runtime/src/proposals_configuration/mod.rs]: As discussed reverted to panic, and updated the comments to reflect actual behavior

**Review by shamil-gadelshin — COMMENTED** [review]
LGTM

**Pull request merged**